lib/ui/painting/image_encoding.cc

```diff
@@ -41,77 +42,78 @@ void InvokeDataCallback(std::unique_ptr<DartPersistentValue> callback,
   }
 }
 
-sk_sp<SkData> EncodeImage(sk_sp<SkImage> image,
-                          SkEncodedImageFormat format,
-                          int quality) {
+sk_sp<SkData> GetImageBytesAsRGBA(sk_sp<SkImage> image) {
   if (image == nullptr) {
     return nullptr;
   }
-  return image->encodeToData(format, quality);
+
+  // Copy the GPU image snapshot into CPU memory.
+  auto cpu_snapshot = image->makeRasterImage();
+  if (!cpu_snapshot) {
+    FXL_LOG(ERROR) << "Pixel copy failed.";
+    return nullptr;
+  }
+
+  SkPixmap pixmap;
+  if (!cpu_snapshot->peekPixels(&pixmap)) {
+    FXL_LOG(ERROR) << "Pixel address is not available.";
+    return nullptr;
+  }
+
+  if (pixmap.colorType() != kRGBA_8888_SkColorType) {
+    // Convert the pixel data to N32 to adhere to our API contract.
+    const auto image_info = SkImageInfo::MakeN32Premul(image->width(),
```

Comment:
Use `MakeRasterData` with a correctly sized allocation and return that allocation directly to save a copy. It's more complication for an edge case for a feature we don't make performance guarantees on. So feel free to keep this version. Your call.